fdbserver/storageserver.actor.cpp

```diff
@@ -590,7 +613,39 @@ struct StorageServer {
 	}
 
 	double getPenalty() {
-		 return std::max(1.0, (queueSize() - (SERVER_KNOBS->TARGET_BYTES_PER_STORAGE_SERVER - 2.0*SERVER_KNOBS->SPRING_BYTES_STORAGE_SERVER)) / SERVER_KNOBS->SPRING_BYTES_STORAGE_SERVER);
+		return std::max(std::min(1.0, (queueSize() - (SERVER_KNOBS->TARGET_BYTES_PER_STORAGE_SERVER -
```

Comment:
The penalty should always be larger than 1. Load balance works by trying to keep an equal number of requests outstanding to all servers, a penalty makes a single outstanding request act like more than one request so you send less traffic to that server. I think the formula you want for penalty would be 1.0/(1.0-currentRate()), and everything should me max instead of having a min.